### PR TITLE
🚧 [PRTL-2561] continuous bin modal

### PR DESCRIPTION
**Pull request by samrichca**
## Ticket Number

PRTL-2561, PRTL-2619, PRTL-2626

## Environment to be used in testing

- [ ] `prod`
- [x] `dev-oicr`
- [ ] `qa` (which version?)

## Description of Changes

Allows the user to create custom continuous bins with custom ranges or intervals

### CODE REVIEW

File: src/packages/@ncigdc/uikit/Modal.js [hunk 1]
```diff
@@ -9,25 +9,26 @@ import './Modal.css';
 ReactModal.setAppElement('#root');
```
Comment (samrichca) [inline src/packages/@ncigdc/uikit/Modal.js]: make the modal fit the page better & alphabetize CSS properties

File: src/packages/@ncigdc/modern_components/ClinicalAnalysis/ClinicalVariableCard/index.js [hunk 2]
```diff
@@ -1484,7 +1557,7 @@ export default compose(
 
         const valuesForPlot =
           variable.plotTypes === 'categorical'
-            ? [...valuesForTable]
+            ? [...filteredData.map(d => d.keyArray).slice(0, 2)]
```
Comment (samrichca) [inline src/packages/@ncigdc/modern_components/ClinicalAnalysis/ClinicalVariableCard/index.js]: this makes survival plots work for custom categorical buckets - it sends in arrays of real property names (e.g. `[male, female, not reported]`) instead of the custom bucket name.

File: src/packages/@ncigdc/modern_components/ClinicalAnalysis/ClinicalVariableCard/index.js [hunk 3]
```diff
@@ -1543,7 +1616,8 @@ export default compose(
   ),
   withPropsOnChange(
     (props, nextProps) => props.variable.active_chart !== nextProps.variable.active_chart ||
-      !isEqual(props.data, nextProps.data),
+      !isEqual(props.data, nextProps.data) ||
+      !isEqual(props.variable.bins, nextProps.variable.bins),
```
Comment (samrichca) [inline src/packages/@ncigdc/modern_components/ClinicalAnalysis/ClinicalVariableCard/index.js]: check if bins changed, i.e. if custom bins were added

File: src/packages/@ncigdc/modern_components/ClinicalAnalysis/ContinuousAggregationQuery.js [hunk 4]
```diff
@@ -27,18 +27,21 @@ const getContinuousAggs = ({ fieldName, stats, filters, bins }) => {
   }
 
   let rangeArr = _.reduce(bins, (acc, bin, key) => {
+    const binValues = bin.key.split('-').map(keyValue => Number(keyValue));
+    const binFrom = binValues[0];
+    const binTo = binValues[1];
     if (
       !!bin &&
```
Comment (samrichca) [inline src/packages/@ncigdc/modern_components/ClinicalAnalysis/ContinuousAggregationQuery.js]: get correct values from the bin

allow buckets that fall outside of available data

File: src/packages/@ncigdc/modern_components/ClinicalAnalysis/ClinicalVariableCard/index.js [hunk 5]
```diff
@@ -1393,53 +1438,85 @@ export default compose(
...
             },
-          ],
-          nextInterval: key,
-        };
+          },
+        ];
       },
     })
   ),
+  withProps(({ data, fieldName, variable }) => {
```
Comment (samrichca) [inline src/packages/@ncigdc/modern_components/ClinicalAnalysis/ClinicalVariableCard/index.js]: create default continuous data for

- resetting buckets
- showing default information in the continuous binning modal

File: src/packages/@ncigdc/modern_components/ClinicalAnalysis/ClinicalVariableCard/index.js [hunk 6]
```diff
@@ -1215,19 +1237,21 @@ const ClinicalVariableCard: React.ComponentType<IVariableCardProps> = ({
...
                           fieldName,
                           id,
-                          value: dataBuckets.reduce((acc, r) => ({
-                            ...acc,
-                            [r.key]: {
-                              ...r,
-                              groupName: r.key,
-                            },
-                          }), {}),
+                          value: variable.plotTypes === 'continuous' ? defaultData.buckets : ([
```
Comment (samrichca) [inline src/packages/@ncigdc/modern_components/ClinicalAnalysis/ClinicalVariableCard/index.js]: use default data to reset the buckets

File: src/packages/@ncigdc/modern_components/ClinicalAnalysis/ClinicalVariableCard/index.js [hunk 7]
```diff
@@ -1175,38 +1174,61 @@ const ClinicalVariableCard: React.ComponentType<IVariableCardProps> = ({
...
-                                id,
-                                value: newBins,
-                                variableKey: 'bins',
-                              }),
-                            );
-                            dispatch(setModal(null));
-                          }
-                          }
-                          />,
+                        variable.plotTypes === 'continuous'
```
Comment (samrichca) [inline src/packages/@ncigdc/modern_components/ClinicalAnalysis/ClinicalVariableCard/index.js]: use a different modal for continuous vs categorical binning

File: src/packages/@ncigdc/modern_components/ClinicalAnalysis/ClinicalVariableCard/index.js [hunk 8]
```diff
@@ -591,9 +589,9 @@ const ClinicalVariableCard: React.ComponentType<IVariableCardProps> = ({
 }) => {
   const tableData = variable.active_chart === 'box'
     ? getBoxTableData(dataValues)
-    : getCategoricalTableData(
-      customBins,
-      getBucketRangesAndFilters,
+    : getTableData(
```
Comment (samrichca) [inline src/packages/@ncigdc/modern_components/ClinicalAnalysis/ClinicalVariableCard/index.js]: renaming things to specify if they're used in continuous or categorical binning, or both

File: src/packages/@ncigdc/modern_components/ClinicalAnalysis/ClinicalVariableCard/index.js [hunk 9]
```diff
@@ -226,15 +227,14 @@ const styles = {
...
+const parseContinuousBucketValue = value => (value % 1
+  ? value
   : Math.round(value * 100) / 100);
 
-const getRangeValue = (key, nextInterval) => `${parseBucketValue(key)}${
-  nextInterval === 0
-    ? ' and up'
-    : ` to ${parseBucketValue(nextInterval - 1)}`
-  }`;
+const getContinuousRangeValue = keyValues => {
```
Comment (samrichca) [inline src/packages/@ncigdc/modern_components/ClinicalAnalysis/ClinicalVariableCard/index.js]: change how values are parsed - integers come back from the API with a `.0` suffix

use integers or up to 2 decimal places, no more `.0`

File: src/packages/@ncigdc/modern_components/ClinicalAnalysis/ClinicalVariableCard/index.js [hunk 10]
```diff
@@ -1359,87 +1391,130 @@ export default compose(
...
                     value: `set_id:${setId}`,
                   },
-                  op: 'in',
                 },
                 {
+                  op: '>=',
                   content: {
                     field: fieldName,
-                    value: [`${parseBucketValue(key)}`],
+                    value: [keyMin],
```
Comment (samrichca) [inline src/packages/@ncigdc/modern_components/ClinicalAnalysis/ClinicalVariableCard/index.js]: change continuous survival filter from interval to range

File: src/packages/@ncigdc/components/Modals/ContinuousBinning/CustomIntervalFields.js [hunk 11]
```diff
@@ -0,0 +1,66 @@
...
+  inputStyles,
+  intervalErrors,
+  intervalFields,
+  validateIntervalFields,
+}) => {
+  return (
+    <div
+      className="binning-interval"
+      onMouseDown={() => {
+        if (disabled) {
```
Comment (samrichca) [inline src/packages/@ncigdc/components/Modals/ContinuousBinning/CustomIntervalFields.js]: &&

File: src/packages/@ncigdc/components/Modals/ContinuousBinning/RangeTableRow.js [hunk 12]
```diff
@@ -0,0 +1,286 @@
...
+
+class RangeTableRow extends React.Component {
+  state = {
+    fieldErrors: defaultFieldState,
+    fieldValues: defaultFieldState,
+  };
+
+  componentDidMount() {
+    const { fields } = this.props;
+    /* eslint-disable */
```
Comment (caravinci) [inline src/packages/@ncigdc/components/Modals/ContinuousBinning/RangeTableRow.js]: why is this necessary?
